config_parser.cc

```diff
@@ -152,6 +152,7 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
   config_stack.push(config);
   TokenType last_token_type = TOKEN_TYPE_START;
   TokenType token_type;
+  bool block_unclosed = false;
```

Comment:
The use of a boolean does not take care of an unbalanced nested statement. For example, with this code, the statement:

Server {
    Sample { 
            }

would return true even though it is unbalanced, because the last close brace changes the block_unclosed variable to false.

Maybe use a counter variable to count the number of unmatched open brackets?